src/main/stgeorge/validator/RequestValidate.java

```diff
@@ -0,0 +1,102 @@
+package stgeorge.validator;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class RequestValidate {
+    private Map<String, String> data;
+
+    public RequestValidate(Map<String, String> data) {
+        this.data = data;
+    }
+
+    public  List<RequestValidationResult> validateRequest()
+    {
+        List<RequestValidationResult> results = new ArrayList<RequestValidationResult>();
+        if(data.isEmpty())
+        {
+            RequestValidationResult result = new RequestValidationResult(false,"Request is Empty");
+            results.add(result);
+            return  results;
+        }
+        if(data.get("firstname").length() <=1)
```

Comment:
Assign 1 to a final const value and re use every were else.

Reply:
If data.get("firstname") is Null then we will get NullPointerException

Reply:
Ok